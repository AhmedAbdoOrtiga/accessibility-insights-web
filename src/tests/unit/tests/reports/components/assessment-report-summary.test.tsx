// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import {
    AssessmentReportSummary,
    AssessmentReportSummaryProps,
} from 'reports/components/assessment-report-summary';

describe('AssessmentReportSummary', () => {
    describe('render', () => {
        test('Reflow UI is disabled', () => {
            const reportData: AssessmentReportSummaryProps = {
                summary: {
                    reportSummaryDetailsData: [],
                },
                featureFlagStoreData: {
                    reflowUI: false,
                } as FeatureFlagStoreData,
            } as AssessmentReportSummaryProps;

            const testObject = new AssessmentReportSummary(reportData);
            const actual = testObject.render();

            expect(actual).toMatchSnapshot();
        });
        test('Reflow UI is enabled', () => {
            const reportData: AssessmentReportSummaryProps = {
                summary: {
                    reportSummaryDetailsData: [],
                },
                featureFlagStoreData: {
                    reflowUI: true,
                } as FeatureFlagStoreData,
            } as AssessmentReportSummaryProps;

            const testObject = new AssessmentReportSummary(reportData);
            const actual = testObject.render();

            expect(actual).toMatchSnapshot();
        });
    });
});
