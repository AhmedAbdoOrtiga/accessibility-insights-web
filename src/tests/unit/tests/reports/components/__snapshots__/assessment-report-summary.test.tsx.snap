// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentReportSummary render Reflow UI is disabled 1`] = `
<div
  className="assessment-report-summary"
>
  <h2>
    Summary
  </h2>
  <OutcomeSummaryBar
    allOutcomeTypes={
      Array [
        "pass",
        "incomplete",
        "fail",
      ]
    }
    countSuffix="%"
  />
  <h3
    className="test-details-text"
  >
    Test details
  </h3>
  <FlaggedComponent
    disableJSXElement={
      <AssessmentSummaryDetails
        testSummaries={Array []}
      />
    }
    enableJSXElement={
      <ReflowAssessmentSummaryDetails
        testSummaries={Array []}
      />
    }
    featureFlag="exportReportOptions"
    featureFlagStoreData={
      Object {
        "reflowUI": false,
      }
    }
  />
</div>
`;

exports[`AssessmentReportSummary render Reflow UI is enabled 1`] = `
<div
  className="assessment-report-summary"
>
  <h2>
    Summary
  </h2>
  <OutcomeSummaryBar
    allOutcomeTypes={
      Array [
        "pass",
        "incomplete",
        "fail",
      ]
    }
    countSuffix="%"
  />
  <h3
    className="test-details-text"
  >
    Test details
  </h3>
  <FlaggedComponent
    disableJSXElement={
      <AssessmentSummaryDetails
        testSummaries={Array []}
      />
    }
    enableJSXElement={
      <ReflowAssessmentSummaryDetails
        testSummaries={Array []}
      />
    }
    featureFlag="exportReportOptions"
    featureFlagStoreData={
      Object {
        "reflowUI": true,
      }
    }
  />
</div>
`;
